### PR TITLE
feat(BILL-CPC-1278): created a customer focused controller

### PR DESCRIPTION
**Pull request by MercuryAtom31**
**JIRA:** link to jira ticket

https://champlainsaintlambert.atlassian.net/browse/CPC-1278

## Context:
What is the ticket about and why are we doing this change.

This Pull Request is the first of three parts. It introduces a new controller class with customer-focused endpoints, ensuring admin-related operations (such as deleting bills) are not exposed to customers. Additions were made to the business layer, and persistence, integration, and unit tests have been implemented. 

Part 2 will focus on the API Gateway, and Part 3 will integrate the backend with the frontend.

## Does this PR change the .vscode folder in petclinic-frontend?:

It does not.

If the PR changes the .vscode folder, explain why in detail because it should not. 
Be sure to include cgerard321 as a reviewer.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
What are the various changes and what other modules do those changes affect.
This can be bullet point or sentence format.

I have changed/created the following classes:

BillService
BillServiceImpl

BillRepository

CustomerBillsController

TESTS:

BillServicePersistenceTests

CustomerBillsControllerIntegrationTest
CustomerBillsControllerUnitTest

Screenshot 2024-09-29 at 19 10 42

Screenshot 2024-09-29 at 19 12 34

Screenshot 2024-09-29 at 19 13 39

Screenshot 2024-09-29 at 19 15 05

BillServicePersistenceTests:
Screenshot 2024-09-29 at 19 15 37

Screenshot 2024-09-29 at 19 16 40

Screenshot 2024-09-29 at 19 16 49

## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 

No changes made to the frontend, thus far.

## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

File: package-lock.json
Comment (Futurespast) [inline package-lock.json:1]: Hmmm, looks like it did the same change on this file like your last pull request.

Comment (MercuryAtom31) [reply package-lock.json:1]: Indeed. To my understanding, this file can be updated by running an "npm" command, such as, "npm install". However, I have not run any npm command!

**Review by Futurespast — APPROVED** [review]
Like I heard in the previous PR if it's just reformatting the it should be fine

**Review by ctsaki25 — APPROVED** [review]
Your code looks proper and your changes are good, dont forget to resolve the package-lock.json conflict (ignore yours and keep the original). I only see one negative controller unit test, make sure to add some more.

**Pull request closed without merging**